e2etest/bgptests/exclude_node.go

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package bgptests
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.universe.tf/e2etest/pkg/k8s"
+	"go.universe.tf/e2etest/pkg/k8sclient"
+	"go.universe.tf/e2etest/pkg/metallb"
+
+	frrconfig "go.universe.tf/e2etest/pkg/frr/config"
+	"go.universe.tf/e2etest/pkg/ipfamily"
+	testservice "go.universe.tf/e2etest/pkg/service"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+var _ = ginkgo.Describe("FRR BGP ExcludeNode IPV4", func() {
+	var (
+		ann           = [2]string{"ocp-mc", "rebooting"}
+		allNodes      *corev1.NodeList
+		excludeCM     = "excludenodepattern" // configmap name is hardcoded to manifests
+		cs            clientset.Interface
+		svc           *corev1.Service
+		testNamespace string
+	)
+
+	restartSpeakerPods := func() {
+		err := metallb.RestartSpeakerPods(cs)
+		Expect(err).NotTo(HaveOccurred())
+		Eventually(func() error {
```

Comment:
how about embedding this waiting logic in `RestartSpeakerPods` ?

Reply:
I would prefer to keep it like that, because I think is cleaner to keep ginkgo{Expect,Eventually} asserts on the test file . If I embed I should use go (not-ginkgo) functions.

Plus that will break the other test using that (Graceful Restart) that expects to return so to check other stuff in between